LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -175,6 +164,37 @@ async Task ProcessJoinRequestAsync(LoRaRequest request)
                 // Build join accept downlink message
                 Array.Reverse(netId);
                 Array.Reverse(appNonceBytes);
+                byte[] dlSettings = new byte[1]
+                {
+                    0
+                };
+
+                if (request.LoRaRegion.RegionLimits.IsCurrentDRIndexWithinAcceptableValue(loRaDevice.DesiredRX2DataRate))
+                {
+                    dlSettings[0] =
+                        (byte)(loRaDevice.DesiredRX2DataRate & 0b00001111);
+                }
+                else
+                {
+                    Logger.Log(devEUI, $"twin RX2 datarate values are not within acceptable values", LogLevel.Error);
+                }
+
+                if (loRaDevice.DesiredRX1DROffset >= 0 && loRaDevice.DesiredRX1DROffset < request.LoRaRegion.RX1DROffsetTable.GetUpperBound(1))
+                {
+                    var rx1droffset = (byte)(loRaDevice.DesiredRX1DROffset << 4);
+                    if (dlSettings == null)
+                    {
+                        dlSettings = new byte[1] { rx1droffset };
+                    }
+                    else
+                    {
+                        dlSettings[0] = (byte)(dlSettings[0] + rx1droffset);
```

Comment:
Is the init here deliberately adding to the existing value? If so, a comment would be good explaining what we do here.